src/Akeneo/Pim/Enrichment/Product/back/Test/Integration/Handler/UpsertProductIntegration.php

```diff
@@ -1107,6 +1110,144 @@ public function it_throws_an_exception_when_currency_does_not_exist(): void
         ]));
     }
 
+    /** @test */
+    public function it_creates_and_updates_a_product_with_a_table_value(): void
+    {
+        $this->createAttribute([
+            'code' => 'a_table',
+            'type' => AttributeTypes::TABLE,
+            'group' => 'other',
+            'localizable' => false,
+            'scopable' => false,
+            'table_configuration' => [
+                [
+                    'code' => 'ingredient',
+                    'data_type' => 'select',
+                    'labels' => [],
+                    'options' => [
+                        ['code' => 'salt'],
+                        ['code' => 'egg'],
+                        ['code' => 'butter'],
+                    ],
+                ],
+                [
+                    'code' => 'quantity',
+                    'data_type' => 'number',
+                    'labels' => [],
+                ],
+            ],
+        ]);
+
+        $product = $this->productRepository->findOneByIdentifier('identifier');
+        Assert::assertNull($product);
+
+        // create product 'identifier'
+        $command = new UpsertProductCommand(
+            userId: $this->getUserId('admin'),
+            productIdentifier: 'identifier',
+            valueUserIntents: [
+                new SetTableValue(
+                    'a_table',
+                    null,
+                    null,
+                    [
+                        ['ingredient' => 'salt'],
+                        ['ingredient' => 'egg', 'quantity' => 2],
+                    ]
+                )
+            ]
+        );
+        $this->messageBus->dispatch($command);
+
+        $this->clearDoctrineUoW();
+
+        $tableNormalizer = new TableNormalizer();
```

Comment:
```suggestion
        $tableNormalizer = $this->get('...');
```

Using the service is better for the future, if we nned to inject something in the class for instance